### PR TITLE
Etcd cluster bug

### PR DESCRIPTION
**Pull request by abdulrahmantkhalifa**

### CODE REVIEW

File: templates/etcd_cluster/actions.py [hunk 1]
```diff
@@ -163,6 +163,8 @@ def watchdog_handler(job):
             dead_nodes.add(node.name)
 
         if len(working_etcds) > (len(etcds)-1)/2:
+            service.model.data.status = 'running'
```
Comment (khaledkbadr) [inline templates/etcd_cluster/actions.py]: add it after the recovering itself

File: pyclient/zeroos/orchestrator/sal/ETCD.py [hunk 2]
```diff
@@ -105,12 +105,17 @@ def start(self):
...
 
     def stop(self):
+        import time
         jobID = "etcd.{}".format(self.name)
         self.container.client.job.kill(jobID)
-        try:
-            self.container.client.job.list(jobID)
-        except RuntimeError:
-            return
+        for i in range(15):
```
Comment (khaledkbadr) [inline pyclient/zeroos/orchestrator/sal/ETCD.py]: see example for waiting in nbdserver actions for cleaner one

File: templates/etcd_cluster/actions.py [hunk 3]
```diff
@@ -175,18 +177,17 @@ def watchdog_handler(job):
...
-        j.tools.async.wrappers.sync(container.executeAction('stop', context=job.context))
-        j.tools.async.wrappers.sync(container.delete())
+        j.tools.async.wrappers.sync(etcd.parent.executeAction('stop', context=job.context))
+
 
     # clean all reaminag tcps on old  running nodes
     for etcd in service.producers['etcd']:
         for tcp in etcd.producers['tcp']:
             try:
+                if not etcd.parent:
```
Comment (khaledkbadr) [inline templates/etcd_cluster/actions.py]: why do we need this?

File: pyclient/zeroos/orchestrator/sal/Container.py [hunk 4]
```diff
@@ -69,8 +74,11 @@ def __init__(self, name, node, flist, hostname=None, mounts=None, nics=None,
                 nic['monitor'] = True
 
     @classmethod
-    def from_containerinfo(cls, containerinfo, node):
+    def from_containerinfo(cls, containerinfo, node, logger=None):
+        if not logger:
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/Container.py]: may be just a one liner `logger = logger or default logger`

File: pyclient/zeroos/orchestrator/sal/Container.py [hunk 5]
```diff
@@ -86,10 +94,13 @@ def from_containerinfo(cls, containerinfo, node):
...
                    arguments['privileged'],
                    arguments['identity'],
-                   arguments['env'])
+                   arguments['env'],
+                   logger=logger)
 
     @classmethod
-    def from_ays(cls, service, password=None):
+    def from_ays(cls, service, password=None, logger=None):
+        if not logger:
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/Container.py]: may be just a one liner `logger = logger or default logger`

File: pyclient/zeroos/orchestrator/sal/ETCD.py [hunk 6]
```diff
@@ -4,20 +4,25 @@
...
 class EtcdCluster:
     """etced server"""
 
-    def __init__(self, name, dialstrings, mgmtdialstrings):
+    def __init__(self, name, dialstrings, mgmtdialstrings, logger=None):
         self.name = name
         self.dialstrings = dialstrings
         self.mgmtdialstrings = mgmtdialstrings
         self._ays = None
+        self.logger = default_logger
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/ETCD.py]: same as first comment

File: pyclient/zeroos/orchestrator/sal/ETCD.py [hunk 7]
```diff
@@ -4,20 +4,25 @@
...
+        if logger:
+            self.logger = logger
 
     @classmethod
-    def from_ays(cls, service, password=None):
+    def from_ays(cls, service, password=None, logger=None):
+        if not logger:
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/ETCD.py]: may be just a one liner `logger = logger or default logger`

File: pyclient/zeroos/orchestrator/sal/StorageCluster.py [hunk 8]
```diff
@@ -24,9 +24,14 @@ def __init__(self, label, nodes=None, disk_type=None):
         self.data_shards = 0
         self.parity_shards = 0
         self._ays = None
+        self.logger = default_logger
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/StorageCluster.py]: same as the first comment

File: pyclient/zeroos/orchestrator/sal/StorageCluster.py [hunk 9]
```diff
@@ -24,9 +24,14 @@ def __init__(self, label, nodes=None, disk_type=None):
...
+        if logger:
+            self.logger = logger
 
     @classmethod
-    def from_ays(cls, service, password):
+    def from_ays(cls, service, password, logger=None):
+        if not logger:
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/StorageCluster.py]: may be just a one liner `logger = logger or default logger`

File: pyclient/zeroos/orchestrator/sal/StorageCluster.py [hunk 10]
```diff
@@ -147,15 +152,18 @@ def __repr__(self):
 class StorageServer:
     """StorageEngine servers"""
 
-    def __init__(self, cluster):
+    def __init__(self, cluster, logger=None):
         self.cluster = cluster
         self.container = None
         self.storageEngine = None
+        self.logger = default_logger
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/StorageCluster.py]: Same as first comment

File: pyclient/zeroos/orchestrator/sal/StorageCluster.py [hunk 11]
```diff
@@ -206,9 +214,12 @@ def __repr__(self):
 
 
 class StorageDashboard:
-    def __init__(self, cluster):
+    def __init__(self, cluster, logger=None):
         self.cluster = cluster
         self.store = 'statsdb'
+        self.logger = default_logger
```
Comment (serboctor) [inline pyclient/zeroos/orchestrator/sal/StorageCluster.py]: Same as first comment

**Pull request merged**